Add docstrings to `optuna.termintor` (#4675)
optuna/optuna: docs/source/reference/terminator.rst

.. module:: optuna.terminator

optuna.terminator
=================

> The :mod:`~optuna.terminator` module implements a mechanism for automatically terminating the optimization process, accompanied by a callback class for the termination and evaluators for the estimated room for improvement in the optimization and statistical error of the objective function. The terminator determines to terminate the optimization process when the improvement is estimated to be smaller than the statistical error.

Reviewer: ```suggestion
The :mod:`~optuna.terminator` module implements a mechanism for automatically terminating the optimization process, accompanied by a callback class for the termination and evaluators for the estimated room for improvement in the optimization and statistical error of the objective function. The terminator stops the optimization process when the estimated potential improvement is smaller than the statistical error.
```
